.github/workflows/ci.yaml

```diff
@@ -0,0 +1,85 @@
+name: CI
+
+on:
+  schedule:
+    - cron: "0 10 * * *"
+  push:
+    branches:
+      - "main"
+    tags:
+      - "v*.*.*"
+  pull_request:
+    branches:
+      - "main"
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+          cache: 'pip'
+
+      - name: Install dependencies
+        run: |
+          pip install -e .
+          pip install "torch>=2.2.2" "numpy>=1.24.4,<2.0.0" "safetensors>=0.4.4" "transformers>=4.44.2" "pytest>=8.3.2" "pytest-depends>=1.0.1"
```

Comment:
@jlamypoirier, would be better if there was a a target for this in `setup.cfg`, wdyt?

Reply:
We could, but isn't [CORE,OPTIONAL,DEV] enough? the remaining dependencies aren't much, and we probably want to test them all eventually.

Reply:
The issue is that .[core,optional,dev] tries to install too much, including things that are cuda only (flash attention). This is just running in the cpu-only free GitHub hosted runner. We will never have GPUs here. It's possible to self-host, but we would have to justify the need to get funding to host this on Toolkit or Azure.

Reply:
Flash attn can be installed in cpu only, see comment in setup.cfg

Reply:
Ok, thanks. I'll add all of these dependencies to the tests, even though not all of them are needed to run them. If this is slowing down the tests too much or introduces other issues, I will ask you to make a change to group the dependencies differently.

Reply:
I'm now trying with `FLASH_ATTENTION_SKIP_CUDA_BUILD=TRUE FLASH_ATTENTION_FORCE_BUILD=TRUE`

Reply:
still no luck: https://github.com/ServiceNow/Fast-LLM/actions/runs/11441529161/job/31829779085?pr=4#step:4:62

Reply:
I can't get it to build.

Reply:
Hmm that worked with torch pre-installed, but here it's installing torch at the same time and the the setup.py seems to be run before torch is installed... One option would be to install torch first in a separate step, otherwise feel free to make a more custom solution.

Reply:
yes, that's what I did: first installing torch, then the rest (including fa). Works now.